apps/meteor/app/livechat/server/api/v1/contact.ts

```diff
@@ -41,29 +42,48 @@ API.v1.addRoute(
 
 API.v1.addRoute(
 	'omnichannel/contact.search',
-	{ authRequired: true },
+	{ authRequired: true, permissionsRequired: ['view-l-room'] },
 	{
 		async get() {
 			check(this.queryParams, {
 				email: Match.Maybe(String),
 				phone: Match.Maybe(String),
+				custom: Match.Maybe(String),
 			});
+			const { email, phone, custom } = this.queryParams;
 
-			const { email, phone } = this.queryParams;
-
-			if (!email && !phone) {
+			if (!email && !phone && !custom) {
 				throw new Meteor.Error('error-invalid-params');
 			}
+			let foundCF = {};
+			if (custom) {
+				const customObj = Object.fromEntries(Array.from(new URLSearchParams(custom)));
+
+				foundCF = Object.fromEntries(
```

Comment:
this logic is being used in two places already, it would be good to have it in a helper function

Reply:
It made less sense after the refactoring into models, but taken into account